logistics_project/apps/malawi/warehouse/warehouse_view.py

```diff
@@ -16,6 +18,27 @@ class MalawiWarehouseView(ReportView):
     
     show_report_nav = True  # override to hide
 
+    # If overridden to True, implement get_min_start_date() and the start date
+    # for the report will automatically update if the default range or the
+    # user-selected range is too wide.
+    automatically_adjust_datespan = False
+
+    def get_min_start_date(self, request):
+        """
+        Should return the minimum start date that should be used with this report.
+        """
+        raise NotImplementedError()
+
+    def update_datespan(self, request):
+        min_date = self.get_min_start_date(request)
+        if min_date and min_date > request.datespan.startdate:
+            startdate = min_date
+            enddate = request.datespan.enddate
+            if enddate < startdate:
+                enddate = startdate
+            request.datespan = DateSpan(startdate, enddate, request.datespan.format)
+            messages.warning(request, "The date range has been automatically adjusted because it was too wide.")
```

Comment:
nit: "too wide" might be a little confusing. maybe "The date range has been automatically adjusted to fit the oldest available data" or something?